### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by LBoleij**
We now have a progressBar 👍

### CODE REVIEW

**Comment by Jipje** [comment]
Using the functionality looks good, will code review now.

File: src/main/java/gui/MenuController.java [hunk 1]
```diff
@@ -278,13 +268,43 @@ private void bookmarked(Button bookmark) {
...
-     * @return the sequence hashMap.
+     * getter for the SequenceMap.
+     * @return The sequenceMap.
      */
     HTreeMap<Long, String> getSequenceHashMap() {
         return fileController.getSequenceHashMap();
     }
 
+    @Override
+    public void update(Observable o, Object arg) {
```
Comment (Jipje) [inline src/main/java/gui/MenuController.java]: Wow good work man

File: src/main/java/gui/sub_controllers/FileController.java [hunk 2]
```diff
@@ -63,6 +74,31 @@ public File chooseFile(Stage stage) {
...
+    public void openFileClicked(GraphicsContext gc, String filePath, MenuController mC)
+            throws IOException, InterruptedException {
+        this.gc = gc;
+        parser = new GfaParser(filePath);
+        parser.addObserver(this);
+        parser.addObserver(mC);
+        if (this.parseThread != null) {
+            this.parseThread.interrupt();
+        }
+        this.parseThread = new Thread(parser);
```
Comment (Jipje) [inline src/main/java/gui/sub_controllers/FileController.java]: Good effort.

**Comment by LBoleij** [comment]
You were right, I accidentally introduced some bugs. They should be fixed by the last 3 commits

**Comment by Jipje** [comment]
Good effort.

**Pull request merged**